libs/aws/odc/aws/queue.py

```diff
@@ -3,6 +3,58 @@
 from typing import Mapping, Any, Iterable, Optional
 
 
+def redrive_queue(
+    queue_name: str,
+    to_queue_name: Optional[str] = None,
+    limit: Optional[int] = None,
+    dryrun: Optional[bool] = False,
+):
+    """
+    Redrive messages from one queue to another. Default usage is to define
+    a "deadletter" queue, and pick its "alive" counterpart, and redrive
+    messages to that queue.
+    """
+    dead_queue = get_queue(queue_name)
+    alive_queue = None
+
+    if to_queue_name is not None:
+        alive_queue = get_queue(to_queue_name)
+    else:
+        count = 0
+        for q in dead_queue.dead_letter_source_queues.all():
+            alive_queue = q
+            count += 1
+            if count > 1:
+                raise Exception(
+                    "Deadletter queue has more than one source, please specify the target queue name."
+                )
+        if count == 0:
+            raise Exception(
+                "No alive queue found for the deadletter queue, please check your configuration."
+            )
```

Comment:
This is better done without iterations
- get a list of source queue candidate
- list has one element -- great use that
- else report error with appropriate message for 0 and >1

```python
qq = list(dead_queue.dead_letter_source_queues.all())
if len(qq) == 0:
   raise ...
elif len(qq) > 1:
   raise ...
assert len(qq) == 1
active_queue = qq[0]
```
```